### ISSUE TITLE
vweb: unable to build in production mode code that injects variables to web templates via $vweb.html()

### ISSUE THREAD

**Issue by smartiniOnGitHub** [vlang/v#8141]
My environment:
OS: linux, Linux Mint 20
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i5-4200U CPU @ 1.60GHz
CC version: cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

getwd: /home/sandro/work/temp
vmodules: /home/sandro/.vmodules
vroot: /opt/vlang_v
vexe: /opt/vlang_v/v
vexe mtime: 2021-01-15 17:56:44
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.1 3397624.1c6fe83

Git version: git version 2.25.1
Git vroot status: 0.2-479-g1c6fe834
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 7543de81

**What did you do?**
Build `examples/vweb/vweb_example.v` in production mode, with:
```
v run -prod vweb_example.v
```

**What did you expect to see?**
Build successful (as it was time ago).
 
**What did you see instead?**
Build errors, because variables injected to web template page seems unused to `$vweb.html()`
, so I get these compiler errors:
```
>>> is_embed_file: false | is_html: true | s:  | n: html | path: ./index.html
vweb_example.v:30:2: error: unused variable: `show`
   28 | pub fn (mut app App) index() vweb.Result {
   29 |     app.cnt++
   30 |     show := true
      |     ~~~~
   31 |     // app.text('Hello world from vweb')
   32 |     hello := 'Hello world from vweb'
vweb_example.v:32:2: error: unused variable: `hello`
   30 |     show := true
   31 |     // app.text('Hello world from vweb')
   32 |     hello := 'Hello world from vweb'
      |     ~~~~~
   33 |     numbers := [1, 2, 3]
   34 |     return $vweb.html()
vweb_example.v:33:2: error: unused variable: `numbers`
   31 |     // app.text('Hello world from vweb')
   32 |     hello := 'Hello world from vweb'
   33 |     numbers := [1, 2, 3]
      |     ~~~~~~~
   34 |     return $vweb.html()
   35 | }
```

**Comment by smartiniOnGitHub** [comment]
Hi all, maybe what do you think on the ability to pass desired variables in the call to `$vweb.html()`, as a workaround, but in general it maybe it could be useful to pass variables (of different types) to the compile time function in an explicit way ... hope this suggestion helps, at least to unblock this.
Thank s a lot.

**Comment by changrui** [comment]
Yes, they are arrors in `-prod` mode instead of `warning` under `v run vweb_examples.v`, maybe fixed.

**Comment by LouisSchmieder** [comment]
this is a bug, I asked @medvednikov already and it get fixed soon

**Comment by LouisSchmieder** [comment]
I'm on a fix in #8180
